packages/jazz-tools/src/tools/coValues/coList.ts

```diff
@@ -521,13 +524,60 @@ export class CoListJazzApi<L extends CoList> extends CoValueJazzApi<L> {
     return getCoValueOwner(this.coList);
   }
 
+  /**
+   * Get an item from the raw CoList, ignoring the query view
+   */
+  private rawGet(rawIndex: number): CoListItem<L> | undefined {
+    const itemDescriptor = this.schema[ItemsSym] as Schema;
+    const rawValue = this.raw.get(rawIndex);
+    if (itemDescriptor === "json") {
+      return rawValue as CoListItem<L> | undefined;
+    } else if ("encoded" in itemDescriptor) {
+      return rawValue === undefined
+        ? undefined
+        : itemDescriptor.encoded.decode(rawValue);
+    } else if (isRefEncoded(itemDescriptor)) {
+      return rawValue === undefined || rawValue === null
+        ? undefined
+        : accessChildByKey(this.coList, rawValue as string, String(rawIndex));
+    }
+  }
+
+  /**
+   * Returns the item at the specified index.
+   *
+   * `coList.$jazz.get(index)` is equivalent to `coList[index]`.
+   *
+   * @param index The index to get the value at.
+   * @returns The item at the specified index, or undefined if the index is out of bounds.
+   * @category Content
+   */
+  get(index: number): CoListItem<L> | undefined {
+    const rawIdx = this.toRawIndex(index);
+    if (rawIdx === undefined) {
+      return undefined;
+    }
+    return this.rawGet(rawIdx);
+  }
+
+  /**
+   * Inserts a value at the specified index.
+   * @param index The index to set the value at.
+   * @param value The value to set.
+   * @category Content
+   */
   set(index: number, value: CoFieldInit<CoListItem<L>>): void {
     const itemDescriptor = this.schema[ItemsSym];
     const rawValue = toRawItems([value], itemDescriptor, this.owner)[0]!;
     if (rawValue === null && !itemDescriptor.optional) {
       throw new Error(`Cannot set required reference ${index} to undefined`);
     }
-    this.raw.replace(index, rawValue);
+    const rawIdx = this.toRawIndex(index);
+    if (rawIdx === undefined) {
+      return;
+    }
+    this.raw.replace(rawIdx, rawValue);
+    this.syncQueryView();
```

Comment:
Do we really need to invalidate the queryView here?

We do it already in SubscriptionScope.updateValue and that should be enough in theory to cover also syncronous local updates.